public/app/features/snapshot/snapshot_ctrl.ts

```diff
@@ -0,0 +1,43 @@
+///<reference path="../../headers/common.d.ts" />
+
+import angular from 'angular';
+import _ from 'lodash';
+
+export class SnapshotsCtrl {
+
+  /** @ngInject */
+  constructor(backendSrv, $scope) {
+    $scope.init = function() {
```

Comment:
yea sorry, forgot to mention that, you will have to use $rootScope to access that function, like this:
 $rootScope.onAppEvent('zoom-out', () => this.zoom(2), $scope);
    
The reason for wanting to write code like this is to ease migration to angular2  (angular2 has completely removed the concept of scopes), code like this is also less coupled trough hidden scope prototype chains